component/loki/write/types.go

```diff
@@ -70,6 +71,21 @@ func (r *EndpointOptions) Validate() error {
 	return nil
 }
 
+// QueueConfig controls how the queue logs remote write client is configured. Note that this client is only used when the
+// loki.write component has WAL support enabled.
+type QueueConfig struct {
+	Capacity     int           `river:"capacity,attr,optional"`
+	DrainTimeout time.Duration `river:"drain_timeout,attr,optional"`
+}
+
+// SetToDefault implements river.Defaulter.
+func (q *QueueConfig) SetToDefault() {
+	*q = QueueConfig{
+		Capacity:     10,
```

Comment:
Mmm true, for some reason I was assuming the default in `prometheus.remote_write` was this, but it's not. I think an initial heuristic could be something like `capacity = buffer capacity in bytes / batch size`, with batch size the one configured [here](https://grafana.com/docs/agent/next/flow/reference/components/loki.write/#endpoint-block). In that way the user could configure this channel size thinking in bytes in memory (obviously assuming a worst case where each batch enqueued is full). wdyt?

Reply:
I think if they can configure it based on overall memory usage thats ideal.

Reply:
Awesome, will implement that then. `loki.write` has a default batch size of 1MiB. I think a default capacity of 100 MiB (which translated in the default case to a channel of size 100) is good. wdyt?

Reply:
That feels a bit high for generic usage. IE the most common case is colocated on a server with other noisy processes. Since it is disk backed what about pulling that back to 10 instead of 100?

Reply:
Yes, for sure

Reply:
Changed to `10MiB`. Once callout is if `buffer capacity in bytes < batch size`, there will be no buffering, and the queue will act similar to the regular client's behaviour (blocking), but I think that's ok. Maybe worth adding a callout in the docs?